book/src/pil/declarations.md

```diff
@@ -0,0 +1,27 @@
+# Declarations
+
+Powdr-pil allows the same syntax to declare various kinds of symbols. This includes
+constants, fixed columns, witness columns and even macros. It deduces the symbol kind
+by its type and the way the symbol is used.
+
+A symbol can be declared using ``let <name>;`` and it can be declared and defined
+using ``let <name> = <value>;``, where ``<value>`` is an expression.
+Symbols without a value are witness columns.
+
+Examples:
+
+```rust
+// This defines a constant
+let rows = 2**16;
+// This defines a fixed column that contains the row number in each row.
+let step = |i| i;
+// Here, we have a witness column.
+let x;
+// This functions returns the square of its input.
+let square = |x| x*x;
+// A recursive function, taking a function and an integer as parameter
+let sum = |f, i| match i {
+    0 => f(0),
+    _ => f(i) + sum(f, i - 1)
+};    
```

Comment:
Wait, can we actually evaluate this?!

Reply:
Sure, why not? I mean when all the pending PRs are merged.